webnn/resources/utils.js

```diff
@@ -289,6 +289,7 @@ const PrecisionMetrics = {
   argMax: {ULP: {int64: 0}},
   argMin: {ULP: {int64: 0}},
   batchNormalization: {ULP: {float32: 6, float16: 6}},
+  cast: {ULP: {float32: 1, float16: 1, int32: 0, uint32: 0, int64: 0, int8: 0, uint8: 0}},
```

Comment:
These are good. Cast is interesting because the ULP *technically* depends on combination of input and the output data types, but I don't know that you can easily represent that via this lookup table (which is why I put 1 ULP in my table).

- int8 -> int32 : 0 ULP
- int32 -> int8 : 0 ULP (assuming value is between -128 and 127)
- int32 -> float32 : 0 ULP (assuming abs(x) <= 16777216)
- float32 -> int32 : 0 ULP
- float16 -> float32 : 0 ULP
- float32 -> float16 : 1 ULP
- float64 -> float32 : 1 ULP (not that WebNN supports float64 anyway)

Ideally we could just say 0 ULP for all cases, but some GPU's round toward zero, while others round toward nearest even. Round to zero is the [D3D specified mode](https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.2%20Floating%20Point%20Conversion), but not all GPU's do so. (we hit this issue for float64 to float32 casting).